docs/source/tutorials/chat.rst

```diff
@@ -0,0 +1,331 @@
+    encode. It simply manages adding all the special tokens in the correct
+    places after encoding the individual messages.
+
+We can see that the tokenizer handled all the formatting without us specifying a prompt
+template. Turns out, all of the additional tags are special tokens, and we don't require
+a separate prompt template. We can verify this by checking if the tags get encoded
+as their own token IDs.
+
+.. code-block:: python
+
+    print(tokenizer._encode_special_token("<|begin_of_text|>"))
+    # 128000
+    print(tokenizer._encode_special_token("<|eot_id|>"))
+    # 128009
+
+The best part is - all these special tokens are handled purely by the tokenizer.
+That means you won't have to worry about messing up any required prompt templates!
+
+
+When should I use a prompt template?
+------------------------------------
+
+When to use or not to use a prompt template is governed by what your desired inference
+behavior is. You should use a prompt template in the following scenarios:
+
+1. You are running inference on the base model and it was pre-trained with a prompt
+template
+2. You want to prime a fine-tuned model to expect a certain prompt structure on inference
+for a specific task
+
+It is not strictly necessary to fine-tune with a prompt template, but generally you
+want the model to perform some sort of task, which will require some formatting of
+the prompt.
+
+For example, the :class:`~torchtune.data.SummarizeTemplate` provides a lightweight
+structure to prime your fine-tuned model for prompts asking to summarize text.
+
+.. code-block:: python
+
+    f"Summarize this dialogue:\n{dialogue}\n---\nSummary:\n"
+
+You can fine-tune LLaMA2 with this template even though the model was originally pre-trained
+with the :class:`~torchtune.data.Llama2ChatFormat`, as long as this is what the model
+sees during inference. The model should be robust enough to adapt to a new template.
+
+
+Fine-tuning on a custom chat dataset
+------------------------------------
+
+Let's test our understanding by trying to fine-tune the LLaMA3-8B model with a custom
+chat dataset. We'll walk through how to set up our data so that it can be tokenized
+correctly and fed into our model.
+
+Let's say we have a local dataset saved as a CSV file that contains questions
+and answers from an online forum. How can we get something like this into a format
+LLaMA3 understands and tokenizes correctly?
+
+.. code-block:: python
+
+    import pandas as pd
+
+    df = pd.read_csv('your_file.csv', nrows=1)
+    print("Header:", df.columns.tolist())
+    # ['input', 'output']
+    print("First row:", df.iloc[0].tolist())
+    # [
+    #     "How do GPS receivers communicate with satellites?",
+    #     "The first thing to know is the communication is one-way...",
+    # ]
+
+The LLaMA3 tokenizer class, :class:`~torchtune.modules.tokenizers.TikTokenTokenizer`,
+expects the input to be in the :class:`~torchtune.data.Message` format. Let's
```

Comment:
You use the `Message` class a few times prior to this. Might be worth a code snippet with a sentence or two explaining everything somewhere in here.

Reply:
nah, I think the docstrings serve that purpose. Going through eot/ipython/etc would be distracting.